File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/java/F04.P2.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Main {
    static int probe() {
        List<Integer> xs = new ArrayList<>(List.of(4, 5, 6));
        // BEGIN PROBE F04.P2
        List<Integer> window = Collections.unmodifiableList(xs);
        return window.get(0);
        // END PROBE F04.P2
    }

    public static void main(String[] args) {
        System.out.println(probe());
    }
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/cpp/F04.P2.cpp
#include <cstdio>
#include <vector>

int probe() {
    std::vector<int> xs = {4, 5, 6};
// BEGIN PROBE F04.P2
const auto& window = xs;
return window[0];
// END PROBE F04.P2
}

int main() {
    std::printf("%d\n", probe());
}
